src/dict.c

```diff
@@ -74,6 +79,30 @@ struct dictEntry {
     struct dictEntry *next; /* Next entry in the same hash bucket. */
 };
 
+typedef struct {
+    union {
+        void *val;
+        uint64_t u64;
+        int64_t s64;
+        double d;
+    } v;
+    struct dictEntry *next; /* Next entry in the same hash bucket. */
+    uint8_t key_header_size; /* offset into key_buf where the key is located at. */
+    unsigned char key_buf[]; /* buffer to embed the key. */
```

Comment:
```suggestion
    unsigned char key_buf[]; /* buffer to embedded key. */
```

Reply:
Isn't it correct `buffer to embed the key.` ?

Reply:
`buffer with embedded key`? The phrasing of `buffer to embed the key` implies that it will later be embeded with a key, but it's always embedded with the key.